buildSrc/src/test/java/org/elasticsearch/gradle/NoticeTaskTest.java

```diff
@@ -0,0 +1,165 @@
+package org.elasticsearch.gradle;
+
+import org.elasticsearch.gradle.test.GradleUnitTestCase;
+import org.gradle.api.Project;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.gradle.testfixtures.ProjectBuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.rules.TemporaryFolder;
+
+import static java.util.Collections.singletonMap;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.apache.commons.io.FileUtils.write;
+import static org.junit.Assert.*;
+
+public class NoticeTaskTest extends GradleUnitTestCase {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    private NoticeTask noticeTask;
+    private Project project;
+    final String outputHeader = "This is the header for the output file\nIt should contain:\n3 lines & 2 spaces";
+
+    public List<File> getListWithCopies() throws IOException{
+        File directory1 = temporaryFolder.newFolder("directoryA");
+        File directory2 = temporaryFolder.newFolder("directoryB");
+
+        File d1Notice = new File(directory1,"test-NOTICE.txt");
+        File d1NoticeCopy = new File(directory2,"test-NOTICE.txt");
+        File d1License = new File(directory1,"test-LICENSE.txt");
+        File d1LicenseCopy = new File(directory2,"test-LICENSE.txt");
+        File d2Notice = new File(directory2,"test2-NOTICE.txt");
+        File d2License = new File(directory2,"test2-LICENSE.txt");
+
+        write(d1Notice,"d1 Notice text file");
+        write(d1NoticeCopy,"d1 Copy Notice text file");
+        write(d2Notice,"d2 Notice text file");
+
+        write(d1License,"d1 License text file");
+        write(d1LicenseCopy,"d1 License text file");
+        write(d2License,"d2 License text file");
+
+
+        List<File> files = new ArrayList<>();
+        files.add(d1License);
+        files.add(d1Notice);
+        files.add(d1LicenseCopy);
+        files.add(d1NoticeCopy);
+        files.add(d2License);
+        files.add(d2Notice);
+
+        return files;
+    }
+
+    public List<File> getListWithoutCopies() throws IOException{
+        List<File> files = this.getListWithCopies();
+        files.remove(2);
```

Comment:
This makes the test hard to read, I would rather create the list without copies here and add to it above.